uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Reminder.php

```diff
@@ -1009,4 +993,77 @@ private static function getUfMatchContact($contact_id) {
     return $uf_match_data;
   }
 
+  /**
+   * Constructs a query condition that excludes "completed" activities.
+   *
+   * @return string
+   */
+  private static function getCompletedActivitiesExclusionQuery() {
+    $documentIncompleteStatuses = implode(',', self::_getDocumentIncompleteStatuses());
+    $documentTypesIds = implode(',', self::_getTypesIdsForComponent('CiviDocument'));
+    $taskIncompleteStatuses = implode(',', self::_getTaskIncompleteStatuses());
+    $taskTypesIds = implode(',', self::_getTypesIdsForComponent('CiviTask'));
+
+    return "(
+      (
+        a.status_id IN ($documentIncompleteStatuses)
+        AND a.activity_type_id IN ($documentTypesIds)
+      )
+      OR
+      (
+        a.status_id IN ($taskIncompleteStatuses)
+        AND a.activity_type_id IN ($taskTypesIds)
+      )
+    )";
+  }
+
+  /**
+   * Returns incomplete statuses for Documents
+   *
+   * @return array
+   */
+  private static function _getDocumentIncompleteStatuses() {
```

Comment:
This function/logic should be defined in the CRM_Tasksassignments_BAO_Document class. Its the class responsibility to expose statuses it deems as completed or not.

Reply:
Yes, I was thinking about that as well. Done.